shared/components/data/store-connection/index.jsx

```diff
@@ -27,34 +29,47 @@ var StoreConnection = React.createClass( {
 		}
 	},
 
-	componentWillUnmount: function() {
+	componentWillUnmount() {
 		this.removeStoreListeners( this.props.stores );
 	},
 
-	addStoreListeners: function( stores ) {
+	addStoreListeners( stores ) {
 		stores.forEach( function( store ) {
 			store.on( 'change', this.handleStoresChanged );
 		}, this );
 	},
 
-	removeStoreListeners: function( stores ) {
+	removeStoreListeners( stores ) {
 		stores.forEach( function( store ) {
 			store.off( 'change', this.handleStoresChanged );
 		}, this );
 	},
 
-	handleStoresChanged: function() {
+	handleStoresChanged() {
 		this.setState( this.props.getStateFromStores( this.props ) );
 	},
 
-	render: function() {
+	isDataLoading() {
+		if ( ! this.props.loadingPlaceholder || ! this.props.isDataLoading ) {
+			return false;
+		}
+
+		return this.props.isDataLoading( this.state );
+	},
+
+	render() {
+		if ( this.isDataLoading() ) {
+			return React.createElement( this.props.loadingPlaceholder );
```

Comment:
What are the benefits of rendering a placeholder component in `StoreConnection`, as opposed to the render() of the child of `StoreConnection` (like `EditCardDetails`)?